python/ray/experimental/gpu_object_manager/gpu_object_store.py

```diff
@@ -101,6 +102,12 @@ def __init__(self):
         self.gpu_object_store: Dict[str, List["torch.Tensor"]] = {}
         # A set of object IDs that are the primary copy.
         self.primary_gpu_object_ids: Set[str] = set()
+        # A dictionary that maps an object ID to the number of readers.
+        # This is used to implement garbage collection for receiver actors,
+        # handling cases where the same GPU object reference is passed to the
+        # same actor task multiple times. For sender actors, we still rely on
+        # the object store's reference counting mechanism.
+        self.gpu_object_id_to_num_readers: Dict[str, int] = {}
```

Comment:
I think we are collecting enough metadata per object that it would be better now to collapse all of the fields for gpu_object_store, primary_gpu_object_ids, gpu_object_id_to_num_readers into one class. It's also better to prevent potential memory leaks where we forget to clean up one dict.

Reply:
good idea! updated